src/features/transfer/TransferTokenForm.tsx

```diff
@@ -255,6 +260,7 @@ function useSelfTokenBalance(tokenRoutes) {
     : route.originTokenAddress;
   const decimals = !route ? STANDARD_TOKEN_DECIMALS : route.decimals;
   const { balance } = useAccountTokenBalance(originChainId, addressForBalance);
+  useOwnerOfErc721(originChainId, tokenAddress, amount);
```

Comment:
This is run for every ERC20 token as well which is wasteful. Pass down isERC721 or move this hook call somewhere else so it can skip if `!isERC721`

Reply:
I've moved it down to the atom level 👍